specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/privateEndpoint.json

```diff
@@ -567,6 +790,117 @@
         }
       },
       "description": "The information of an AvailablePrivateEndpointType."
+    },
+    "PrivateDNSZoneGroup": {
+      "properties": {
+        "name": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Name of the resource that is unique within a resource group. This name can be used to access the resource."
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource type."
+        },
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/PrivateDNSZoneGroupProperties",
+          "description": "Properties of the private dns zone group."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./network.json#/definitions/SubResource"
+        }
+      ],
+      "description": "Private dns zone group resource."
+    },
+    "PrivateDNSZoneGroupProperties": {
+      "properties": {
+        "provisioningState": {
+          "readOnly": true,
+          "$ref": "./network.json#/definitions/ProvisioningState",
+          "description": "The provisioning state of the private dns zone group resource."
+        },
+        "privateDnsZoneConfigs": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/PrivateDnsZoneConfig"
+          },
+          "description": "A collection of private dns zone configurations of the private dns zone group."
+        }
+      },
+      "description": "Properties of the private dns zone group."
+    },
+    "PrivateDnsZoneConfig": {
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The resource name."
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The resource type."
+        },
```

Comment:
privateDnsZoneConfigs are not their own resource.  They are an object inside an array inside privateDnsZoneGroups.  They shouldn't have their own name, type, and ID.  If they need that they should be a child resource of privateDnsZoneGroups

Reply:
Hey @pilor, do you mean the privateDnsZoneConfig resources? according to the example JSON I've seen: 
"{
    "name": "dnsgroup1",
    "id": "/subscriptions/a9c98ed2-1a82-446b-9ef6-dbb935df6614/resourceGroups/group 1/providers/Microsoft.Network/privateEndpoints/pe_zone/privateDnsZoneGroups/dnsgroup1",
    "type": "Microsoft.Network/privateEndpoints/privateDnsZoneGroups",
    "properties": {
      "provisioningState": "Succeeded",
      "privateDnsZoneConfigs": [
        {
          "name": "config1",
          "id": "/subscriptions/a9c98ed2-1a82-446b-9ef6-dbb935df6614/resourceGroups/group 1/providers/Microsoft.Network/privateEndpoints/pe_zone/privateDnsZoneGroups/dnsgroup1/privateDnsZoneConfigs/config1",
          "type": "Microsoft.Network/privateEndpoints/privateDnsZoneGroups/privateDnsZoneConfigs",
          "properties": {
            "provisioningState": "Succeeded",
            "privateDnsZoneId": "/subscriptions/a9c98ed2-1a82-446b-9ef6-dbb935df6614/resourceGroups/group 1/providers/Microsoft.Network/privateDnsZones/zone1.com",
            "recordSets": [
              {
                "recordType": "A",
                "recordSetName": "abc",
                "fqdn": "abc.zone1.com",
                "ipAddresses": [
                  "10.0.0.4"
                ]
              },
              {
                "recordType": "A",
                "recordSetName": "abc2",
                "fqdn": "abc2.zone1.com",
                "ipAddresses": [
                  "10.0.0.5"
                ]
              }
            ]
          }
        },
        {
          "name": "config2",
          "id": "/subscriptions/a9c98ed2-1a82-446b-9ef6-dbb935df6614/resourceGroups/group 1/providers/Microsoft.Network/privateEndpoints/pe_zone/privateDnsZoneGroups/dnsgroup1/privateDnsZoneConfigs/config2",
          "type": "Microsoft.Network/privateEndpoints/privateDnsZoneGroups/privateDnsZoneConfigs",
          "properties": {
            "provisioningState": "Succeeded",
            "privateDnsZoneId": "/subscriptions/a9c98ed2-1a82-446b-9ef6-dbb935df6614/resourceGroups/group 1/providers/Microsoft.Network/privateDnsZones/zone2.com",
            "recordSets": [
              {
                "recordType": "A",
                "recordSetName": "abc",
                "fqdn": "abc.zone2.com",
                "ipAddresses": [
                  "10.0.0.6"
                ]
              },
              {
                "recordType": "A",
                "recordSetName": "abc2",
                "fqdn": "abc2.zone2.com",
                "ipAddresses": [
                  "10.0.0.7"
                ]
              }
            ]
          }
        }
      ]
    }
  }
"
It looks like they need those attributes (name type and id), so would you happen to know how to specify resources as child resources? Sorry for the confusion if I've caused any.

Reply:
If you want a child resource it should only be available at a URI under the parent and not as an array inside the parent.  Your options are either to have the data as an array in the parent (in which case it wouldn't have its own resource ID, type, RBAC, async provisioning, etc...) or model it as a true child resource at its own URI underneath privateDnsZoneGroups.  privateDnsZoneGroups are themselves a child resource of privateEndpoints so the pattern should be clear.

Reply:
So I suppose this means supporting CRUD operations +list as well for privateDnsZoneConfigs? @ajinkyapotdarvjti I think this makes sense let me know what you think.

Reply:
Looks like that's not a resource we support those operations on, so I think I'll be going the array non child route